### PR TITLE
Add puppet_use_cache parameters

### PR DESCRIPTION
**Pull request by ekohl**
Closes GH-168

Also updates the tests a bit.

### CODE REVIEW

File: manifests/init.pp [hunk 1]
```diff
@@ -346,7 +346,9 @@
   $registered_proxy_url       = $foreman_proxy::params::registered_proxy_url,
   $oauth_effective_user       = $foreman_proxy::params::oauth_effective_user,
   $oauth_consumer_key         = $foreman_proxy::params::oauth_consumer_key,
-  $oauth_consumer_secret      = $foreman_proxy::params::oauth_consumer_secret
+  $oauth_consumer_secret      = $foreman_proxy::params::oauth_consumer_secret,
+  $puppet_use_cache           = $foreman_proxy::params::puppet_use_cache,
+  $puppet_cache_location      = $foreman_proxy::params::puppet_cache_location,
```
Comment (mmoll) [inline manifests/init.pp]: The parameters need documentation in the header for kafo/foreman-installer.

File: manifests/params.pp [hunk 2]
```diff
@@ -225,4 +225,7 @@
       $plugin_prefix = 'smart_proxy_'
     }
   }
+
+  $puppet_use_cache = undef
+  $puppet_cache_location = '/var/lib/smart-proxy/cache'
```
Comment (mmoll) [inline manifests/params.pp]: Packaging does not bring that directory (at least on my DEB installation), so we might have to think of something in the packaging also. Personally I'd put a directory like ```/var/tmp/foreman-proxy/cache``` in place for that, but I'm unsure if that's really the correct place.

Comment (ekohl) [inline manifests/params.pp]: I just copied it from the puppet.yml.example. Maybe `/var/cache/foreman-proxy` would be a better place, but we should standardize on it.

File: manifests/init.pp (hunk 1 above)
Comment (ekohl) [inline manifests/init.pp]: Good point.

File: manifests/params.pp (hunk 2 above)
Comment (mmoll) [inline manifests/params.pp]: :+1:  for ```/var/cache/foreman-proxy```. IMHO we should just create this as empty directory belonging to the foreman-proxy user in the RPM and DEB packages.

Comment (ekohl) [inline manifests/params.pp]: @domcleal agreed?

Comment (ekohl) [inline manifests/params.pp]: Changed and let puppet create the directory so we don't have to wait for new packages.

File: manifests/init.pp (hunk 1 above)
Comment (ekohl) [inline manifests/init.pp]: Done.

**Comment by ekohl** [comment]
Now with passing tests again.

File: spec/classes/foreman_proxy__config__spec.rb [hunk 3]
```diff
@@ -764,4 +751,23 @@
...
+
+  context 'with puppet use_cache enabled' do
+    let :pre_condition do
+      'class {"foreman_proxy":
+        puppet_use_cache => true,
+      }'
+    end
+
+    it 'should create the cache_location' do
+      contain_file('/var/cache/Foreman_proxy').with_ensure('directory')
```
Comment (mmoll) [inline spec/classes/foreman_proxy__config__spec.rb]: shouldn't this be '/var/cache/foreman-proxy'?

Comment (ekohl) [inline spec/classes/foreman_proxy__config__spec.rb]: Not sure how that passed, but you're right.

File: spec/classes/foreman_proxy__config__spec.rb [hunk 4]
```diff
@@ -764,4 +751,23 @@
...
+
+  context 'with puppet use_cache enabled' do
+    let :pre_condition do
+      'class {"foreman_proxy":
+        puppet_use_cache => true,
+      }'
+    end
+
+    it 'should create the cache_location' do
+      contain_file('/var/cache/foreman_proxy').with_ensure('directory')
```
Comment (mmoll) [inline spec/classes/foreman_proxy__config__spec.rb]: I really don't get why the tests are green, I still think there should be a dash and no underscore :)

Comment (ekohl) [inline spec/classes/foreman_proxy__config__spec.rb]: I forgot the should. Also explains why the uppercase failed to fail ;)

File: spec/classes/foreman_proxy__config__spec.rb (hunk 3 above)
Comment (mmoll) [inline spec/classes/foreman_proxy__config__spec.rb]: oh wow :)

**Comment by mmoll** [comment]
merged, thanks @ekohl!

**Comment by ekohl** [comment]
It took more attempts than should have been needed :)

**Pull request merged**